examples/LM_Bundle_Adjust.ipynb

```diff
@@ -0,0 +1,725 @@
+    "##############\n",
+    "\n",
+    "column_dict = compute_coefficient_columns(cnet, sensors, parameters)\n",
+    "num_parameters = max(col_range[1] for col_range in column_dict.values())\n",
+    "num_observations = 2 * len(cnet)\n",
+    "W_observations = np.eye(num_observations)\n",
+    "W_params = compute_parameter_weights(cnet, sensors, parameters, column_dict)\n",
+    "\n",
+    "iteration = 0\n",
+    "V = compute_residuals(cnet, sensors)\n",
+    "dX = np.zeros(W_params.shape[0]) #initialize for sigma calculatio\n",
+    "sigma0 = compute_sigma(V, dX, W_params, W_observations)\n",
+    "print(f'iteration {iteration}: sigma0 = {sigma0}\\n')\n",
+    "\n",
+    "max_iterations = 100\n",
+    "tol = 1e-10\n",
+    "total_correction = np.zeros(num_parameters)\n",
+    "damping = 0.1\n",
+    "for i in range(max_iterations):   \n",
+    "    iteration += 1\n",
+    "    old_sigma0 = sigma0\n",
+    "    \n",
+    "    J = compute_jacobian(cnet, sensors, parameters, column_dict)    \n",
+    "    N = J.T.dot(W_observations).dot(J) + W_params # calculate the normal equation\n",
+    "    C = J.T.dot(W_observations).dot(V) - W_params.dot(total_correction)\n",
+    "    dX = np.linalg.inv(N).dot(C) # calculate change in camera parameters and ground points\n",
+    "    \n",
+    "\n",
+    "    total_correction += dX\n",
+    "    print(f'corrections: mean = {dX.mean()} min = {dX.min()} max = {dX.max()}')\n",
+    "    \n",
+    "    update_parameters(sensors, parameters, cnet, dX, column_dict)\n",
+    "    \n",
+    "    V = compute_residuals(cnet, sensors)\n",
+    "    sigma0 = compute_sigma(V, dX, W_params, W_observations)\n",
+    "#     sigma0 = np.sqrt((V.dot(W_observations).dot(V) + dX.dot(W_params).dot(dX))/dof)\n",
+    "    if (sigma0 < old_sigma0): #good, do we really want to make the step size smaller? \n",
+    "        damping*=0.1\n",
+    "    else:\n",
+    "        damping*=10\n",
+    "        \n",
+    "    print(f'iteration {iteration}: sigma0 = {sigma0}\\n')\n",
+    "    \n",
+    "    if (abs(sigma0 - old_sigma0) < tol):\n",
+    "        print(f'change in sigma0 of {abs(sigma0 - old_sigma0)} converged!')\n",
+    "        break\n",
+    "    "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Whole (Dense) Bundle Process in a loop with LM terms\n",
+    "\n",
+    "The next three cells show examples of the bundle adjustment loop running with Levenberg-Marquat added.\n",
+    "\n",
+    "This example shows the first heuristic for initializing $\\lambda$, the weighting parameter, from Numerical Recipes in C++\n",
+    "\n",
+    "NumericalRecipes in C++ recommended starting with $\\lambda$ = 0.01, and then increasing or decreasing this value by an order of magnitude each iteration depending on whether the last iteration increased or decreased \n",
+    "$\\sigma_0$.\n",
```

Comment:
Do you have a link for where they suggest this?

Reply:
The place I actually found this was a hard-copy book I borrowed from Mary, I'll look for an online source.

Reply:
I was mainly suggesting it for documentation purposes, but based on the conversation we had earlier.. @jessemapel @kberryUSGS should we maybe not have this method in the notebook in the repo?